service_worker.js

```diff
@@ -317,3 +321,35 @@ function logStorageChange(changes, area) {
   }
 }
 chrome.storage.onChanged.addListener(logStorageChange);
+
+
+async function clearOldCacheBackground(tabId) {
+  if (!(await doesTabExist(tabId))) {
+    chrome.storage.local.remove([tabId]);
+  };
+}
+
+async function clearOldCache() {
+  const now = Date.now();
+  chrome.storage.local.get(null, results => {
+    for (let hash in results) {
+      if (results[hash].timestamp) {
+        // If it's a timestamp, check if still valid
+        const timestamp = new Date(results[hash].timestamp).getTime();
+        if (now - timestamp > ONE_DAY_MS ) {
+          console.log('Removing', hash, results[hash]);
+          chrome.storage.local.remove([hash]);
+        }
+      } else if (results[hash] === true || results[hash] === false) {
```

Comment:
nit: kind of confusing check. `typeof results[hash] === 'boolean'` communicates the intent a little more directly, maybe?